shopfloor_base/controllers/main.py

```diff
@@ -25,10 +32,25 @@ def _process_endpoint(
         collection = collection or request.env["shopfloor.app"].browse(app_id)
         # TODO: in base_rest `*args` is passed based on
         # the type of route (eg: /<int:id>/update)
+        params = kwargs
```

Comment:
would you mind adding the link to the change here?

Reply:
@simahawk In Odoo < 16, all the requests to a route defined on a RestController was wrapped into a [HttpRestRequest](https://github.com/OCA/rest-framework/blob/14.0/base_rest/http.py#L119C7-L119C22) by a patch of the [Root.get_request](https://github.com/OCA/rest-framework/blob/14.0/base_rest/http.py#L237)
As side effect the json document was part of the  kwargs passed to the `_process_endpoint`method (https://github.com/OCA/rest-framework/blob/14.0/base_rest/http.py#L125) This behavior is changed into odoo > 16 by the registration of a dedicated dispatcher class to use for the routes declared with a routing type 'restapi' (https://github.com/OCA/rest-framework/blob/16.0/base_rest/http.py#L129). Since you bypass the dispatching made by base_rest and redefine your own handlers for the declared rest methods, the specific dispatcher is no more used. An other fix would be to register your handers with routing_type 'restapi'...